docs/azure/sdk/resource-management.md

```diff
@@ -203,53 +237,52 @@ using System.Threading.Tasks;
 
 // Code omitted for brevity
 
-string resourceId = "/subscriptions/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/resourceGroups/workshop2021-rg/providers/Microsoft.Compute/availabilitySets/ws2021availSet";
-ResourceIdentifier id = new ResourceIdentifier(resourceId);
+ResourceIdentifior resourceId = AvailabilitySetResource.CreateResourceIdentifier("aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee","resourceGroupName","resourceName"); 
 // We construct a new armClient to work with
 ArmClient armClient = new ArmClient(new DefaultAzureCredential());
 // Next we get the specific subscription this resource belongs to
-Subscription subscription = await armClient.GetSubscriptions().GetAsync(id.SubscriptionId);
+SubscriptionResource subscription = await armClient.GetSubscriptions().GetAsync(resourceId.SubscriptionId);
 // Next we get the specific resource group this resource belongs to
-ResourceGroup resourceGroup = await subscription.GetResourceGroups().GetAsync(id.ResourceGroupName);
+ResourceGroupResource resourceGroup = (await subscription.GetResourceGroups().GetAsync(resourceId.ResourceGroupName)).Value;
 // Finally we get the resource itself
 // Note: for this last step in this example, Azure.ResourceManager.Compute is needed
-AvailabilitySet availabilitySet = await resourceGroup.GetAvailabilitySets().GetAsync(id.Name);
+AvailabilitySetResource availabilitySet = (await resourceGroup.GetAvailabilitySets().GetAsync(resourceId.Name)).Value;
 ```
 
-This approach required a lot of code and three API calls to Azure. The same can be done with less code and without any API calls by using extension methods that we've provided on the client itself. These extension methods allow you to pass in a resource identifier and retrieve a scoped resource client. The object returned is a [Resource](#resourcecs). Since it hasn't reached out to Azure to retrieve the data yet, the `Data` property will be null.
+This approach required a lot of code and three API calls to Azure. The same can be done with less code and without any API calls by using extension methods that we've provided on the client itself. These extension methods allow you to pass in a resource identifier and retrieve a scoped resource client. The object returned is a [ResourceName]Resource(#resourcenameresourcecs). Since it hasn't reached out to Azure to retrieve the data yet, the `Data` property will be null.
```

Comment:
Is this `[ResourceName]Resource(#resourcenameresourcecs)` supposed to be a link? It is not in correct format here